orange/src/main/java/com/hartwig/hmftools/orange/OrangeWgsConfig.java

```diff
@@ -0,0 +1,193 @@
+        configBuilder.addPath(CUPPA_DIR_CFG, false, CUPPA_DIR_DESC);
+        configBuilder.addPath(PEACH_DIR_CFG, false, PEACH_DIR_DESC);
+        configBuilder.addPath(SIGS_DIR_CFG, false, SIGS_DIR_DESC);
+    }
+
+    // params for WGS Tumor only
+    @NotNull
+    String annotatedVirusTsv();
+
+    @NotNull
+    String chordPredictionTxt();
+
+    @NotNull
+    String cuppaResultCsv();
+
+    @NotNull
+    String cuppaSummaryPlot();
+
+    @Nullable
+    String cuppaFeaturePlot();
+
+    @Nullable
+    String cuppaChartPlot();
+
+    @NotNull
+    String sigsAllocationTsv();
+
+    // additional params for WGS Ref
+    @Nullable
+    String referenceSampleId();
+
+    @Nullable
+    String sageGermlineGeneCoverageTsv();
+
+    @Nullable
+    String sageSomaticRefSampleBQRPlot();
+
+    @Nullable
+    String linxGermlineDataDirectory();
+
+    @Nullable
+    String peachGenotypeTsv();
+
+    @Nullable
+    String refSampleWGSMetricsFile();
+
+    @Nullable
+    String refSampleFlagstatFile();
+
+    @NotNull
+    static OrangeWgsConfig createConfig(@NotNull ConfigBuilder configBuilder)
+    {
+        ImmutableOrangeWgsConfig.Builder builder = ImmutableOrangeWgsConfig.builder();
+        String tumorSampleId = configBuilder.getValue(TUMOR_SAMPLE_ID);
+
+        String pipelineSampleRootDir = checkAddDirSeparator(configBuilder.getValue(PIPELINE_SAMPLE_ROOT_DIR));
+        String sampleDataDir = checkAddDirSeparator(configBuilder.getValue(SAMPLE_DATA_DIR_CFG));
+
+        // params required for WGS, Tumor only
+        String virusDir = getToolDirectory(configBuilder, pipelineSampleRootDir, sampleDataDir, VIRUS_DIR_CFG, VIRUS_INTERPRETER_DIR);
+        if(virusDir == null)
+        {
+            throw new IllegalArgumentException("Virus dir not found but required for WGS config");
+        }
+        builder.annotatedVirusTsv(fileIfExists(AnnotatedVirusFile.generateFileName(virusDir, tumorSampleId)));
+
+        String chordDir = getToolDirectory(configBuilder, pipelineSampleRootDir, sampleDataDir, CHORD_DIR_CFG, CHORD_DIR);
+        if(chordDir == null)
+        {
+            throw new IllegalArgumentException("Chord dir not found but required for WGS config");
+        }
+        builder.chordPredictionTxt(fileIfExists(ChordDataFile.generateFilename(chordDir, tumorSampleId)));
+
+        String cuppaDir = getToolDirectory(configBuilder, pipelineSampleRootDir, sampleDataDir, CUPPA_DIR_CFG, CUPPA_DIR);
+        if(cuppaDir == null)
+        {
+            throw new IllegalArgumentException("Cuppa dir not found but required for WGS config");
+        }
+
+        builder.cuppaResultCsv(fileIfExists(CuppaDataFile.generateFilename(cuppaDir, tumorSampleId)));
+        builder.cuppaSummaryPlot(fileIfExists(CuppaDataFile.generateReportSummaryPlotFilename(cuppaDir, tumorSampleId)));
+        builder.cuppaFeaturePlot(optionalFileIfExists(CuppaDataFile.generateReportFeaturesPlotFilename(cuppaDir, tumorSampleId)));
```

Comment:
This is correct, but one note for clarification: Cuppa either fits on one page or on two pages. If Cuppa requires 2 pages it generates a second file called 'feature plot'. In case CUPPA fits on one page, the feature plot is included in the summary plot. That's why the feature plot is truly optional (like linx plot dir)